restdocs-api-spec-openapi3-generator/src/test/kotlin/com/epages/restdocs/apispec/openapi3/OpenApi3GeneratorTest.kt

```diff
@@ -324,6 +324,73 @@ class OpenApi3GeneratorTest {
         thenOpenApiSpecIsValid()
     }
 
+    @Test
+    fun `should fail for enum values request parameter with wrong type`() {
+        givenResourcesWithRequestParameterWithWrongEnumValues()
+
+        assertThrows<NumberFormatException> { whenOpenApiObjectGenerated() }
+    }
+
+    @Test
+    fun `should fail for enum values header parameter with wrong type`() {
+        givenResourcesWithHeaderParameterWithWrongEnumValues()
+
+        assertThrows<NumberFormatException> { whenOpenApiObjectGenerated() }
+    }
+
+    @Test
+    fun `should include enum values`() {
+        givenResourcesWithEnumValues()
+
+        whenOpenApiObjectGenerated()
+
+        val params = openApiJsonPathContext.read<List<Map<*, *>>>("paths./products/{id}.get.parameters.*")
+
+        then(params).anyMatch {
+            it["name"] == "booleanParameter" &&
+                (it["schema"] as LinkedHashMap<*, *>)["type"] == "boolean" &&
+                (it["schema"] as LinkedHashMap<*, *>)["enum"] == listOf(true, false)
+        }
+        then(params).anyMatch {
+            it["name"] == "stringParameter" &&
+                (it["schema"] as LinkedHashMap<*, *>)["type"] == "string" &&
+                (it["schema"] as LinkedHashMap<*, *>)["enum"] == listOf("PV1", "PV2", "PV3")
+        }
+        then(params).anyMatch {
+            it["name"] == "numberParameter" &&
+                (it["schema"] as LinkedHashMap<*, *>)["type"] == "number" &&
+                (it["schema"] as LinkedHashMap<*, *>)["enum"] == listOf(1, 2, 3)
+        }
+        then(params).anyMatch {
+            it["name"] == "integerParameter" &&
+                (it["schema"] as LinkedHashMap<*, *>)["type"] == "integer" &&
+                (it["schema"] as LinkedHashMap<*, *>)["enum"] == listOf(1, 2, 3)
+        }
+        then(params).anyMatch {
+            it["name"] == "X-SOME-BOOLEAN" &&
+                (it["schema"] as LinkedHashMap<*, *>)["type"] == "boolean" &&
+                (it["schema"] as LinkedHashMap<*, *>)["enum"] == listOf(true, false)
+        }
+        then(params).anyMatch {
+            it["name"] == "X-SOME-STRING" &&
+                (it["schema"] as LinkedHashMap<*, *>)["type"] == "string" &&
+                (it["schema"] as LinkedHashMap<*, *>)["enum"] == listOf("HV1", "HV2")
+        }
+        then(params).anyMatch {
+            it["name"] == "X-SOME-NUMBER" &&
+                (it["schema"] as LinkedHashMap<*, *>)["type"] == "number" &&
+                (it["schema"] as LinkedHashMap<*, *>)["enum"] == listOf(1, 2, 3)
+        }
+        then(params).anyMatch {
+            it["name"] == "X-SOME-INTEGER" &&
+                (it["schema"] as LinkedHashMap<*, *>)["type"] == "integer" &&
+                (it["schema"] as LinkedHashMap<*, *>)["enum"] == listOf(1, 2, 3)
+        }
+        then(params).hasSize(9)
```

Comment:
It's 9 descriptors due to 4 headers, 4 parameters in the request and 1 header in the response (4+4+1)?